File: Randomi_Go_Test1/Randomi_Go_Test1/Deck.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Randomi_Go_Test1
{
    internal class Deck
    {
        private static List<Cards> deck;

        static Deck()
        {
            InitializeDeck();
        }

        public static void InitializeDeck()
        {
            CreateDeckTest();
            ShuffleDeck();
        }

        private static void CreateDeckTest()
        {
            deck = new List<Cards>
            {
                new Cards(Guid.NewGuid(), "Boule de feu", Element.Feu, Stars.Deux, Nature.Sort, "Enlevez 4 points à votre adversaire", 4, 0, 0, false),
                new Cards(Guid.NewGuid(), "Fléchette", Element.Air, Stars.Un, Nature.Arme, "Enlevez 3 points à votre adversaire", 3, 0, 0, false),
                new Cards(Guid.NewGuid(), "Bulle", Element.Eau, Stars.Un, Nature.Sort, "Lancez un dé, enlevez le résultat à votre adversaire", 0, 0, 0, true),
                new Cards(Guid.NewGuid(), "Pioche", Element.Terre, Stars.Deux, Nature.Sort, "Enlevez 6 points à votre adversaire, piochez une carte", 6, 0, 1, false),
                new Cards(Guid.NewGuid(), "Airicot", Element.Air, Stars.Trois, Nature.Aliment, "Récupérez 15 points", 0, 15, 0, false),
                new Cards(Guid.NewGuid(), "Forage", Element.Terre, Stars.Cinq, Nature.Sort, "Enlevez 9 points à votre adversaire, piochez 3 cartes", 9, 0, 3, false),
                new Cards(Guid.NewGuid(), "Glace à l'eau", Element.Eau, Stars.Trois, Nature.Aliment, "Récupérez 7 points, piochez 2 cartes", 0, 7, 2, false),
                new Cards(Guid.NewGuid(), "Grain de poussière", Element.Terre, Stars.Un, Nature.Sort, "Enlevez 0 point à votre adversaire", 0, 0, 0, false),
                new Cards(Guid.NewGuid(), "Lance de flammes", Element.Feu, Stars.Cinq, Nature.Arme, "Enlevez 18 points à votre adversaire", 18, 0, 0, false),
                new Cards(Guid.NewGuid(), "Bille de feu", Element.Feu, Stars.Un, Nature.Sort, "Enlevez 1 point à votre adversaire", 1, 0, 0, false),
                new Cards(Guid.NewGuid(), "Petit sablé", Element.Terre, Stars.Deux, Nature.Aliment, "Récupérez 5 points", 0, 5, 0, false),
                new Cards(Guid.NewGuid(), "Pistolet à eau", Element.Eau, Stars.Deux, Nature.Arme, "Enlevez 7 points à votre adversaire", 7, 0, 0, false),
                new Cards(Guid.NewGuid(), "Pomme", Element.Terre, Stars.Deux, Nature.Aliment, "Récupérez 6 points", 0, 6, 0, false),
                new Cards(Guid.NewGuid(), "Zéphyr", Element.Air, Stars.Deux, Nature.Sort, "Enlevez 5 points à votre adversaire, piochez une carte", 5, 0, 1, false),
                
            };
        }

        public static List<Cards> GetDeck()
        {
            return new List<Cards>(deck);
        }

        public static void ShuffleDeck()
        {
            Random rng = new Random();
            int n = deck.Count;
            while (n > 1)
            {
                n--;
                int k = rng.Next(n + 1);
                (deck[n], deck[k]) = (deck[k], deck[n]); // Swap des cartes
            }
        }

        public static Cards DrawCard()
        {
            if (deck.Count != 0)
            {
                Cards drawnCard = deck[0];
                deck.RemoveAt(0);
                return drawnCard;
            }
            else
            {
                return null;


            }
        }
    }
}

File: gameengine/src/deck.rs
//gestion des differentes actions
use crate::card;
use rand::random;
use rand::thread_rng;
use rand::prelude::SliceRandom;
use crate::card::Card;
//creer un deck
pub fn create_deck()
{
    println!("création d'un deck");

}
//faire une methode qui renvoie une liste de carte

pub fn create_deck_test() -> Vec<Card>
{
    
    //ajouter la carte dans une liste de carte
    //exemple de carte
    //creer une liste de carte
    let mut deck: Vec<card::Card> = Vec::new();
    //ajouter la carte dans le deck
    deck.push(card::Card::new(
        String::from("Boule de feu"),
        card::Element::Fire,
        card::Stars::Two,
        card::Kind::Spell,
        String::from("Inflige 4 points de dégâts."),
        4,
        0,
        0,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Pomme"),
        card::Element::Earth,
        card::Stars::Two,
        card::Kind::Food,
        String::from("Récupérez 6 points."),
        0,
        6,
        0,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Pêche"),
        card::Element::Water,
        card::Stars::Two,
        card::Kind::Food,
        String::from("Récupérez 4 points. Piochez 1 carte."),
        0,
        4,
        1,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Arbalète"),
        card::Element::Air,
        card::Stars::Four,
        card::Kind::Weapon,
        String::from("Enlevez 9 points de vie à votre adversaire. Piochez 1 carte."),
        9,
        0,
        1,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Bille de feu"),
        card::Element::Fire,
        card::Stars::One,
        card::Kind::Spell,
        String::from("Enlevez 1 point de vie à votre adversaire. "),
        1,
        0,
        0,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Bulle"),
        card::Element::Water,
        card::Stars::One,
        card::Kind::Spell,
        String::from("Lancé un dé. Enlevez le résultat du dé à votre adversaire."),
        0,
        0,
        0,
        true,
    ));
    deck.push(card::Card::new(
        String::from("Pioche"),
        card::Element::Earth,
        card::Stars::Two,
        card::Kind::Weapon,
        String::from("Enlevez 6 points de vie à votre adversaire. Piochez 1 carte."),
        6,
        0,
        1,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Pomme d'amour"),
        card::Element::Fire,
        card::Stars::Two,
        card::Kind::Food,
        String::from("Vous ainsi qu'un autre joueur récupérez 12 points de vie."),
        0,
        12,
        0,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Fléchettes"),
        card::Element::Air,
        card::Stars::One,
        card::Kind::Weapon,
        String::from("Enlevez 3 points de vie à votre adversaire. Piochez 1 carte."),
        3,
        0,
        1,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Grain de poussière"),
        card::Element::Earth,
        card::Stars::One,
        card::Kind::Spell,
        String::from("Enlevez 0 point de vie à votre adversaire."),
        3,
        0,
        1,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Noix de coco"),
        card::Element::Water,
        card::Stars::Four,
        card::Kind::Food,
        String::from("Récupérez 4 points et enlevez 9 points de vie à votre adversaire."),
        9,
        4,
        0,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Epuisette"),
        card::Element::Water,
        card::Stars::Two,
        card::Kind::Weapon,
        String::from("Enlevez 2 points à votre adversaire. Piochez 2 carte."),
        2,
        0,
        2,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Bouteille d'eau"),
        card::Element::Water,
        card::Stars::Three,
        card::Kind::Weapon,
        String::from("Récupérez 8 points. Enlevez 4 points à votre adversaire. Piochez 2 carte."),
        4,
        8,
        2,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Lance de flammes"),
        card::Element::Fire,
        card::Stars::Five,
        card::Kind::Weapon,
        String::from("Enlevez 18 points de vie à votre adversaire."),
        18,
        0,
        0,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Terrassement"),
        card::Element::Earth,
        card::Stars::Five,
        card::Kind::Spell,
        String::from("Enlevez 15 points de vie à votre adversaire. Piochez 2 cartes."),
        15,
        0,
        2,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Vague vampirique"),
        card::Element::Water,
        card::Stars::Five,
        card::Kind::Spell,
        String::from("Enlevez 12 points de vie à votre adversaire. Récupérez 6 points."),
        12,
        6,
        0,
        false,
    ));
    deck.push(card::Card::new(
        String::from("Rafale"),
        card::Element::Air,
        card::Stars::Five,
        card::Kind::Spell,
        String::from("Enlevez 14 points de vie à votre adversaire. Piochez 3 cartes."),
        14,
        0,
        3,
        false,
    ));
    for i in 0..deck.len() {
        println!("Carte {} : {:?}", i + 1, deck[i]);
       //afficher 
    }
    return deck;
}
//melanger un deck
pub fn shuffle_deck(deck: &mut Vec<Card>) {
    let mut rng = thread_rng();
    deck.shuffle(&mut rng);
    println!("melanger le deck");
}

//piocher une carte du deck
pub fn draw_card()
{
    println!("piocher une carte du deck");
}